interpreter/src/lib.rs

```diff
@@ -0,0 +1,352 @@
+// Copyright (C) 2019-2024 Aleo Systems Inc.
+// This file is part of the Leo library.
+
+// The Leo library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Leo library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the Leo library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    collections::HashMap,
+    fmt::Display,
+    fs,
+    path::{Path, PathBuf},
+};
+
+use colored::*;
+
+use snarkvm::prelude::TestnetV0;
+
+use leo_ast::{Ast, Node as _, NodeBuilder};
+
+use leo_span::{Span, source_map::FileName, symbol::with_session_globals};
+
+use leo_errors::{CompilerError, InterpreterHalt, LeoError, Result, emitter::Handler};
+
+mod cursor;
+use cursor::*;
+
+/// Contains the state of interpretation, in the form of the `Cursor`,
+/// as well as information needed to interact with the user, like
+/// the breakpoints.
+pub struct Interpreter {
+    cursor: Cursor<'static>,
+    actions: Vec<InterpreterAction>,
+    handler: Handler,
+    node_builder: NodeBuilder,
+    breakpoints: Vec<Breakpoint>,
+    filename_to_program: HashMap<PathBuf, String>,
+}
+
+#[derive(Clone, Debug)]
+pub struct Breakpoint {
+    program: String,
+    line: usize,
+}
+
+#[derive(Clone, Debug)]
+pub enum InterpreterAction {
+    LeoInterpretInto(String),
+    LeoInterpretOver(String),
+    RunFuture(usize),
+    Breakpoint(Breakpoint),
+    Into,
+    Over,
+    Step,
+    Run,
```

Comment:
We'll prolly wanna throw in the following commands to manipulate the "global" context.
- Set mapping value
- Get mapping value
- Remove mapping value
- Set block height
- Set caller
- Set signer

Reply:
Oh wait, would users just write leo syntax for the mapping ops?